akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala

```diff
@@ -11,7 +11,7 @@ import akka.event.AddressTerminatedTopic
 
 private[akka] trait DeathWatch { this: ActorCell ⇒
 
-  private var watching: Set[ActorRef] = ActorCell.emptyActorRefSet
+  private var watching: Map[ActorRef, Option[Any]] = Map.empty
```

Comment:
wrapping the collection in another object means another reference variable, i.e. 4 bytes :-P
might work with value class